### PR TITLE
Fixed incorrect string concatenation

### PR DESCRIPTION
**Pull request by epidemia**

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
All changes look good to me though the original approach is not incorrect. It is "just" ineffective

File: test/src/test/java/hudson/ProcTest.java [hunk 1]
```diff
@@ -57,14 +57,15 @@ public void run() {
 
         RemoteLauncher launcher = new RemoteLauncher(StreamTaskListener.NULL, ch, true);
 
-        String str="";
+        StringBuilder stringBuilder = new StringBuilder();
+
         for (int i=0; i<256; i++)
-            str += "oxox";
+            stringBuilder.append("oxox");
```
Comment (daniel-beck) [inline test/src/test/java/hudson/ProcTest.java]: Is the Java compiler really not smart enough to optimize this away?

Comment (oleg-nenashev) [inline test/src/test/java/hudson/ProcTest.java]: JVM may be smart enough, but at least OpenJDK does not optimize it IIRC

**Comment by oleg-nenashev** [comment]
Retriggering the PR builder

File: test/src/test/java/hudson/ProcTest.java [hunk 2]
```diff
@@ -57,14 +57,15 @@ public void run() {
...
 
         for (int i=0; i<1000; i++) {
             ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            launcher.launch().cmds("echo",str).stdout(baos).join();
-            assertEquals(str, baos.toString().trim());
+            launcher.launch().cmds("echo",stringBuilder.toString()).stdout(baos).join();
+            assertEquals(stringBuilder, baos.toString().trim());
```
Comment (oleg-nenashev) [inline test/src/test/java/hudson/ProcTest.java]: It causes a test failure:

screen shot 2017-03-17 at 14 07 31

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
1 errorneous change in the tests

**Comment by oleg-nenashev** [comment]
@epidemia gentle ping

File: test/src/test/java/hudson/ProcTest.java (hunk 2 above)
Comment (mc1arke) [inline test/src/test/java/hudson/ProcTest.java]: Failure is caused by comparing a String to a StringBuilder. Need to call `toString()` on the String builder to make this pass.

**Pull request closed without merging**